lib/Transforms/Scalar/DxilRemoveUnstructuredLoopExits.cpp

```diff
@@ -625,3 +716,48 @@ bool hlsl::RemoveUnstructuredLoopExits(
 
   return changed;
 }
+
+// This pass runs hlsl::RemoveUnstructuredLoopExits.
+// It is used for testing, and can be run from `opt` via option -dxil-r-u-l-e
+namespace {
+
+class DxilRemoveUnstructuredLoopExits : public LoopPass {
+public:
+  static char ID;
+  DxilRemoveUnstructuredLoopExits() : LoopPass(ID) {
+    initializeDxilRemoveUnstructuredLoopExitsPass(
+        *PassRegistry::getPassRegistry());
+  }
+
+  StringRef getPassName() const override {
+    return "Dxil Remove Unstructured Loop Exits";
+  }
+
+  void getAnalysisUsage(AnalysisUsage &AU) const override {
+    AU.addRequired<LoopInfoWrapperPass>();
+    AU.addRequired<DominatorTreeWrapperPass>();
+    AU.addRequiredID(&LCSSAID);
+    // Don't assume it's in LoopSimplifyForm. That is not guaranteed
+    // by the usual callers.
+  }
+
+  bool runOnLoop(Loop *L, LPPassManager &LPM) override {
+    LoopInfo *LI = &getAnalysis<LoopInfoWrapperPass>().getLoopInfo();
+    DominatorTree *DT = &getAnalysis<DominatorTreeWrapperPass>().getDomTree();
+    return hlsl::RemoveUnstructuredLoopExits(L, LI, DT);
+  }
+};
+} // namespace
+
+char DxilRemoveUnstructuredLoopExits::ID;
+
+Pass *llvm::createDxilRemoveUnstructuredLoopExitsPass() {
+  return new DxilRemoveUnstructuredLoopExits();
+}
+
+INITIALIZE_PASS_BEGIN(DxilRemoveUnstructuredLoopExits, "dxil-r-u-l-e",
```

Comment:
Missed opportunity to call this `dxil-rules` :)

A minor point, but I think actually spelling it out (i.e. `dxil-remove-unstructured-loop-exits`) would be a bit clearer to read. It shouldn't be something we have to type often so while it is unwieldly I don't think it would be a big burden.

Reply:
Pretty quickly I started writing shell scripts to run various things; I agree.  :-)

Reply:
Done.